cmd/nexctl/local_unix.go

```diff
@@ -265,6 +266,13 @@ func init() {
 			},
 		},
 	})
+	// Apply privilege check only for 'nexctl nexd' subcommand
+	if len(os.Args) > 1 && os.Args[1] == "nexd" {
+		if err := hasPrivileges(); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	}
```

Comment:
something like this might be a little cleaner, by hooking into the CLI lib:

```
diff --git a/cmd/nexctl/local_unix.go b/cmd/nexctl/local_unix.go
index 6a873d06..0cae0f88 100644
--- a/cmd/nexctl/local_unix.go
+++ b/cmd/nexctl/local_unix.go
@@ -25,6 +25,13 @@ func init() {
                                Required:    false,
                        },
                },
+               Before: func(cCtx *cli.Context) error {
+                       if err := hasPrivileges(); err != nil {
+                               fmt.Println(err)
+                               return err
+                       }
+                       return nil
+               },
                Subcommands: []*cli.Command{
                        {
                                Name:   "version",
```